app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterDetailView.kt

```diff
@@ -0,0 +1,56 @@
+package com.puzzlebench.clean_marvel_kotlin.presentation.mvp.view
+
+import android.view.View
+import com.puzzlebench.clean_marvel_kotlin.R
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
+import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.fragment.CharacterDetailFragment
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.character_detail.*
+import java.lang.ref.WeakReference
+
+class CharacterDetailView(fragment: CharacterDetailFragment) {
+    private val fragmentRef = WeakReference(fragment)
+
+    fun init() {
+        val fragment = fragmentRef.get()
+        if (fragment != null) {
+           showLoading()
+        }
+    }
+
+    fun showToastNoDetailToShow() {
+        val fragment = fragmentRef.get()
+        if (fragment != null) {
+            val message = fragment.activity.baseContext.resources.getString(R.string.no_description_available)
+            fragment.activity.applicationContext.showToast(message)
+        }
+    }
+
+    fun showToastNetworkError(error: String) {
+        fragmentRef.get()?.activity?.applicationContext?.showToast(error)
+    }
+
+    fun showCharacterDetail(character: Character) {
+        val fragment = fragmentRef.get()
+        if (fragment != null) {
```

Comment:
you can replace this by 
fragment ?: and the if body
in this way, you are using kotlin like java and it is not the idea

Reply:
done